### PR TITLE
(maint) Remove unneccesary querying macro to improve streaming syntax

### PR DESCRIPTION
**Pull request by ajroetker**
This commit removes the `with-query-results-cursor` macro, as it was
functionally unused. The macro wrapped `with-query-results-cursor*` with a
different interface, but the only thing that used the macro was
`streamed-query-results` which wrapped the macro in the original
interface of `with-query-results-cursor*`. This commit collapses these
functions/macros.
This commit also cleans up some syntax in jdbc, removing unnecessary
parens.

### CODE REVIEW

File: src/puppetlabs/puppetdb/jdbc.clj [hunk 1]
```diff
@@ -298,38 +300,30 @@
 
   The cursor is closed when `func` returns. If an exception is thrown,
   the query is cancelled."
-  [func sql params]
+  [[sql & params] func]
```
Comment (mullr) [inline src/puppetlabs/puppetdb/jdbc.clj]: I see how it reduces duplication, but this is a very strange function signature.

Comment (ajroetker) [inline src/puppetlabs/puppetdb/jdbc.clj]: It's the same as streamed-query-results was before

Comment (mullr) [inline src/puppetlabs/puppetdb/jdbc.clj]: then it was strange before too. but we should leave it.

File: src/puppetlabs/puppetdb/query/reports.clj [hunk 2]
```diff
@@ -84,13 +83,10 @@
...
   [version url-prefix query-sql]
   {:pre [(map? query-sql)]}
-  (let [{[sql & params] :results-query
-         count-query    :count-query} query-sql
-         result {:result (query/streamed-query-result
-                          version sql params
-                          ;; The doall simply forces the seq to be traversed
-                          ;; fully.
-                          (comp doall (munge-result-rows version url-prefix)))}]
+  (let [{:keys [count-query results-query]} query-sql
```
Comment (mullr) [inline src/puppetlabs/puppetdb/query/reports.clj]: these functions are all the same, aren't they. we should clean that up at some point.

File: src/puppetlabs/puppetdb/jdbc.clj [hunk 3]
```diff
@@ -288,7 +290,7 @@
      (fn []
        ~@body)))
 
-(defn with-query-results-cursor*
+(defn with-query-results-cursor
```
Comment (mullr) [inline src/puppetlabs/puppetdb/jdbc.clj]: I think we should call this streamed-query-result now; it makes more sense at the call site.

File: src/puppetlabs/puppetdb/jdbc.clj (hunk 1 above)
Comment (mullr) [inline src/puppetlabs/puppetdb/jdbc.clj]: ah, there were two overloads before; the one that takes the vector was actually unused.

**Comment by mullr** [comment]
generally looks ok, just some minor comments

File: src/puppetlabs/puppetdb/jdbc.clj (hunk 1 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/jdbc.clj]: No it was used by the query-eng, this is the java.jdbc syntax,

File: src/puppetlabs/puppetdb/query/reports.clj (hunk 2 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/reports.clj]: They are the old query-eng fn's afaik, old code that needs to stick around as long as some of our old endpoints are alive

File: src/puppetlabs/puppetdb/jdbc.clj (hunk 3 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/jdbc.clj]: Maybe we could do that in a separate PR, I kind of like it the way it is honestly, streamed-query-results isn't really similar to the java.jdbc functions

File: src/puppetlabs/puppetdb/query/reports.clj (hunk 2 above)
Comment (mullr) [inline src/puppetlabs/puppetdb/query/reports.clj]: I mean that they all look the same structurally, and could use a refactor in the future.

File: src/puppetlabs/puppetdb/jdbc.clj (hunk 3 above)
Comment (mullr) [inline src/puppetlabs/puppetdb/jdbc.clj]: Totally fine with it being in a different PR. My only real objection is that 'with-blah' usually is a macro, so it's a little surprising to see it as a function which takes a callback.

File: src/puppetlabs/puppetdb/query/reports.clj (hunk 2 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/query/reports.clj]: Agreed :)

File: src/puppetlabs/puppetdb/jdbc.clj (hunk 3 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/jdbc.clj]: Ah wasn't sure about that, yeah how about I make a ticket to rename it and refactor the code from your other comment, that way the refactor will save some renaming hell?

Comment (mullr) [inline src/puppetlabs/puppetdb/jdbc.clj]: Sounds good.

**Comment by ajroetker** [comment]
@mullr Also I know this is maint but it relates the hack-day project I wanted to work on... transducers, so I would be really grateful if we could get this merged by tomorrow :D

**Comment by mullr** [comment]
alright, :+1: from me then

**Comment by wkalt** [comment]
looks good to me
